### ISSUE TITLE
Missing release body 

### ISSUE THREAD

**Issue by sondrelg** [axodotdev/cargo-dist#394]
Hi!

When generating my `dist-manifest.json` I'm unable to make cargo-dist construct a release body based on my changelog. The `announcement_github_body` field remains empty.

## What I've tried
To generate the dist-manifest, I run 
```
cargo dist manifest --tag=<my-tag> --artifacts=all --output-format=json > dist-manifest.json
```

I have tried moving the `CHANGELOG.md` both to my crate directory, and to the workspace directory, and I've tried naming it `RELEASES.md`. 

## Project structure
My virtual workspace looks like this:

```
|__ apps 
|    |
|    |__ cli app
|           |__ Cargo.toml
|
|__ workspace Cargo.toml
```

my workspace Cargo.toml contains:

```toml
[workspace.metadata.dist]
cargo-dist-version = "0.2.0"
ci = ["github"]
targets = [
    "x86_64-unknown-linux-gnu",
    "x86_64-apple-darwin",
    "aarch64-apple-darwin",
]
```

and my app Cargo.toml specifies `publish = true`. 

Publishing artifacts works - I'm just missing the release body.

------------

For a virtual workspace setup like this, is there anything specific I need to do that I haven't covered above? 

Here's my generated dist-manifest.json:
<details>
<summary>show</summary>
<code>
{
  "dist_version": "0.2.0",
  "announcement_tag": "<tag>",
  "announcement_is_prerelease": false,
  "announcement_title": "<tag>",
  "announcement_github_body": "",
  "system_info": {
    "cargo_version_line": "cargo 1.72.0 (103a7ff2e 2023-08-15)"
  },
  "releases": [
    {
      "app_name": "app",
      "app_version": "<version>",
      "artifacts": [
        "app-aarch64-apple-darwin.tar.xz",
        "app-aarch64-apple-darwin.tar.xz.sha256",
        "app-x86_64-apple-darwin.tar.xz",
        "app-x86_64-apple-darwin.tar.xz.sha256",
        "app-x86_64-unknown-linux-gnu.tar.xz",
        "app-x86_64-unknown-linux-gnu.tar.xz.sha256"
      ]
    }
  ],
  "artifacts": {
    "app-aarch64-apple-darwin.tar.xz": {
      "name": "app-aarch64-apple-darwin.tar.xz",
      "kind": "executable-zip",
      "target_triples": [
        "aarch64-apple-darwin"
      ],
      "path": "/home/user/repos/project/target/distrib/app-aarch64-apple-darwin.tar.xz",
      "assets": [
        {
          "name": "CHANGELOG.md",
          "path": "CHANGELOG.md",
          "kind": "changelog"
        },
        {
          "name": "LICENSE",
          "path": "LICENSE",
          "kind": "license"
        },
        {
          "name": "README.md",
          "path": "README.md",
          "kind": "readme"
        },
        {
          "name": "app",
          "path": "app",
          "kind": "executable"
        }
      ],
      "checksum": "app-aarch64-apple-darwin.tar.xz.sha256"
    },
    "app-aarch64-apple-darwin.tar.xz.sha256": {
      "name": "app-aarch64-apple-darwin.tar.xz.sha256",
      "kind": "checksum",
      "target_triples": [
        "aarch64-apple-darwin"
      ],
      "path": "/home/user/repos/project/target/distrib/app-aarch64-apple-darwin.tar.xz.sha256"
    },
    "app-x86_64-apple-darwin.tar.xz": {
      "name": "app-x86_64-apple-darwin.tar.xz",
      "kind": "executable-zip",
      "target_triples": [
        "x86_64-apple-darwin"
      ],
      "path": "/home/user/repos/project/target/distrib/app-x86_64-apple-darwin.tar.xz",
      "assets": [
        {
          "name": "CHANGELOG.md",
          "path": "CHANGELOG.md",
          "kind": "changelog"
        },
        {
          "name": "LICENSE",
          "path": "LICENSE",
          "kind": "license"
        },
        {
          "name": "README.md",
          "path": "README.md",
          "kind": "readme"
        },
        {
          "name": "app",
          "path": "app",
          "kind": "executable"
        }
      ],
      "checksum": "app-x86_64-apple-darwin.tar.xz.sha256"
    },
    "app-x86_64-apple-darwin.tar.xz.sha256": {
      "name": "app-x86_64-apple-darwin.tar.xz.sha256",
      "kind": "checksum",
      "target_triples": [
        "x86_64-apple-darwin"
      ],
      "path": "/home/user/repos/project/target/distrib/app-x86_64-apple-darwin.tar.xz.sha256"
    },
    "app-x86_64-unknown-linux-gnu.tar.xz": {
      "name": "app-x86_64-unknown-linux-gnu.tar.xz",
      "kind": "executable-zip",
      "target_triples": [
        "x86_64-unknown-linux-gnu"
      ],
      "path": "/home/user/repos/project/target/distrib/app-x86_64-unknown-linux-gnu.tar.xz",
      "assets": [
        {
          "name": "CHANGELOG.md",
          "path": "CHANGELOG.md",
          "kind": "changelog"
        },
        {
          "name": "LICENSE",
          "path": "LICENSE",
          "kind": "license"
        },
        {
          "name": "README.md",
          "path": "README.md",
          "kind": "readme"
        },
        {
          "name": "app",
          "path": "app",
          "kind": "executable"
        }
      ],
      "checksum": "app-x86_64-unknown-linux-gnu.tar.xz.sha256"
    },
    "app-x86_64-unknown-linux-gnu.tar.xz.sha256": {
      "name": "app-x86_64-unknown-linux-gnu.tar.xz.sha256",
      "kind": "checksum",
      "target_triples": [
        "x86_64-unknown-linux-gnu"
      ],
      "path": "/home/user/repos/project/target/distrib/app-x86_64-unknown-linux-gnu.tar.xz.sha256"
    }
  }
}
</code>
</details>

**Comment by sondrelg** [comment]
Running it with `--verbose trace` I see the log output contains:

```
 INFO not announcing a consistent version, skipping changelog generation
```

**Comment by sondrelg** [comment]
I use 

```
cargo dist manifest --tag=${{ github.ref_name }} --artifacts=all --no-local-paths --output-format=json > dist-manifest.json
```

In my workflow, which is triggered from pushing a tag like this
```
git tag appname-v0.0.0
git push origin main --tags
```

**Comment by Gankra** [comment]
Hey there! So the relevant lines of code are these ones:

https://github.com/axodotdev/cargo-dist/blob/7d179eae7e00b9f8763c9109dff374c90c55cb08/cargo-dist/src/tasks.rs#L2306-L2308

https://github.com/axodotdev/cargo-dist/blob/7d179eae7e00b9f8763c9109dff374c90c55cb08/cargo-dist/src/changelog.rs#L23-L25

----

The issue you're running into is that cargo-dist is *intentionally* failing to generate your changelog because you're exercising a path we didn't have a good example of, and didn't want to do something weird and broken with. Basically if someone is using the "individual package" release model (`--tag=appname-v0.0.0`), it's sketchy for us to try to apply that logic to a root RELEASES.md because those might be the versions for a different app in the workspace. 

I will gladly properly enable this functionality if you could detail to me the way you *want* it to work. In particular:

> I have tried moving the CHANGELOG.md both to my crate directory, and to the workspace directory, and I've tried naming it RELEASES.md.

Where did you originally have your CHANGELOG(s)? Did you have one changelog file for each crate in your workspace?

**Comment by Gankra** [comment]
Oh right, when #393 lands (a couple hours) we can just ask axoproject for a "package's changelog", and if that matches the behaviour you want, we're golden.

The behaviour is to use a CHANGELOG/RELEASES file in the same directory as the Cargo.toml of each package, falling back to "inheriting" the root workspace one.

**Comment by Gankra** [comment]
fixed by #393
